apps/wallet-mobile/src/features/Receive/common/ReceiveProvider.tsx

```diff
@@ -37,47 +34,45 @@ export const ReceiveProvider = ({
 const receiveReducer = (state: ReceiveState, action: ReceiveAction) => {
   return produce(state, (draft) => {
     switch (action.type) {
-      case ReceiveActionType.CurrentAddressChanged:
+      case ReceiveActionType.SelectedAddressChanged:
         draft.selectedAddress = action.address
         break
 
       default:
-        throw new Error(`invalid action`)
+        throw new Error('[ReceiverContext] invalid action')
     }
   })
 }
 
-export type ReceiveType = 'single' | 'multiple'
-
-type ReceiveAction = {type: ReceiveActionType.CurrentAddressChanged; address: string}
+type ReceiveAction = {type: ReceiveActionType.SelectedAddressChanged; address: string}
 
 export type ReceiveState = {
-  selectedAddress: null | string
-  defaultAddress: null | string
+  selectedAddress: string
+  defaultAddress: string
 }
 
 type ReceiveActions = {
-  currentAddressChanged: (address: string) => void
+  selectedAddressChanged: (address: string) => void
 }
 
 const defaultState: ReceiveState = Object.freeze({
-  selectedAddress: null,
-  defaultAddress: null,
+  selectedAddress: '',
```

Comment:
receive address can't never be null, since is created on wallet basis

Reply:
cant be typed in this way?